src/devices/video/k051316.cpp

```diff
@@ -228,6 +229,9 @@ u8 k051316_device::rom_r(offs_t offset)
 
 void k051316_device::ctrl_w(offs_t offset, u8 data)
 {
+	if ((offset == 0x0e) && (data != m_ctrlram[0x0e]))
```

Comment:
`m_ctrlram` is indeed legacy code. And without a mask on bits 1-2 this will have performance issue if a given client pings the other bits, making these three (including the ROM readback flag) explicit booleans sounds like a good idea to me.